fix: resolve timeout and performance issues in buffer write mode and AOF writes

EquentR opened the pull request:
fix: resolve timeout and performance issues in buffer write mode and AOF writes
- resolve timeout issue after small-batch writes in buffer mode
- address high disk usage and write bottleneck caused by forced AOF file sync

- 解决redis_writer缓冲区模式下小批量写入后的超时不提交问题
- 解决强制AOF文件同步导致的磁盘占用高和写入瓶颈问题

EquentR committed:
fix: resolve timeout and performance issues in buffer write mode and AOF writes
- resolve timeout issue after small-batch writes in buffer mode
- address high disk usage and write bottleneck caused by forced AOF file sync

suxb201 commented on internal/client/redis.go:
```diff
@@ -204,16 +204,16 @@ func (r *Redis) SendBytesBuff(buf []byte) {
 	r.timer.Reset(time.Second)
+	if atomic.AddUint64(&r.sendCount, 1)%100 != 0 {
+		return
+	}
```
新的写法最坏会有 100s 的延迟：每次计时器快结束的时候被重置为 1s，直到第 100 次出发点 %100==0，然后 flush。

EquentR replied:
原先的写法如果写入不到100条，那永远不会flush，这里用了timer而不是ticker，只会执行一次🤣，而且main中与目的端保持的ping也会被cache而没发过去，我再研究一下，感觉两种方法都有问题

suxb201 reviewed (changes requested):
1. 更倾向于原先写法一点，新的改动带来高延迟风险。
2. 后续有机会可以把 sendCount 改成 sendBytes，flush 不在意 count，只在意 bytes；也能避免 100 条大命令导致 OOM 的场景。

EquentR committed:
feat: optimize Redis client buffer flush logic
- Trigger flush based on 64k data size instead of packet count
- Improve timer reset mechanism to minimize latency under high load

feat: enhance RDB file receiving logs
- Add log to display the current received RDB size in real-time
